### DISCUSSION TITLE
Dashboard Link URL should pickup on base url used

### DISCUSSION THREAD

**Discussion by timguy99** [teslamate-org/teslamate#302] [Ideas]
I have my site setup on a ip based url for example something like http://192.168.99.22:4000

Clicking the dashboard link on this page attempts to go to http://127.0.0.1:3000. It should use the base URL that the site on port :4000 is referenced at.

I believe I found a number of other links that similarly do the same so any and all links really should use this method and not assume things are at 127.0.0.1

**Comment by adriankumpf** [comment]
TeslaMate makes an educated guess to get the base url and the url of your grafana installation. It won't always be right. Both URLs can be changed on the settings page.

**Comment by timguy99** [comment]
Shouldn't (or couldn't) it just use relative URL's vs absolute?

**Comment by adriankumpf** [comment]
It is not always the case that the web interface and Grafana are reachable under the same domain/ IP address. What advantage do you see if relative URLs would also be supported?

**Comment by timguy99** [comment]
Maybe an option for "use relative"? Or maybe assume relative unless something is defined in settings?

I'm thinking there is no easy way for my setup cause I hit a 192.168.x.x address when on the local LAN. But if I'm remote, I SSH and setup a tunnel which means I'm hitting it at a 127.0.0.1 address.

**Comment by adriankumpf** [comment]
I see. That makes sense.

**Comment by wxwty168** [comment]
Hey, is it possiable to use baseURL to set url right now?

**Comment by kid1412621** [comment]
really need this. My case is using `telsamate.example.com` via Cloudflare tunnel, and using `teslamate.machine.lan` on local. Grafana would be `${baseURL}/grafana`. Currently seems no way to set up the URLs to meet my needs.

**Reply by timguy99** [reply]
Yeah, I'm still waiting for some change here.

I'm also using Cloudflare now as well and no easy way to accommodate this I don't think. Unless there were / are some changes that I'm aware of.
